client/src/lib/api.ts

```diff
@@ -110,3 +110,22 @@ export async function submitApplication(
   if (!res.ok) throw new Error(`fetch Error ${res.status}`);
   return await res.json();
 }
+
+export async function fetchApplications(
+  userId: number
+): Promise<(Application & Job)[]> {
+  const auth = localStorage.getItem('designengineer');
+  let a;
+  if (auth) {
+    a = JSON.parse(auth);
+  }
```

Comment:
This is fine, but I would put accessing the token in a separate function, since you'll likely want to do it in several places as the app evolves. You could put both the `setItem` and `getItem` functionality into separate functions (perhaps in a separate file?) and export them, so all the logic about dealing with localStorage is in one location.

The fact that 'designengineer' is in two places (at least) should indicate the need to encapsulate this code.

You could either have `readAuth` and `writeAuth` functions or you could even have `readUser`, `readToken`, `writeUser`, `writeToken` functions.